File: src/main/java/world/guild/GuildMark.java
package world.guild;

import lombok.Getter;
import org.jooq.Record;
import util.packet.PacketReader;
import util.packet.PacketWriter;

import static database.jooq.Tables.GUILDMARK;

@Getter
public class GuildMark {

    private final short markBg, mark;
    private final byte markBgColor, markColor;

    public GuildMark(PacketReader reader) {
        markBg = reader.readShort();
        markBgColor = reader.readByte();
        mark = reader.readShort();
        markColor = reader.readByte();
    }

    public GuildMark(Record rec) {
        this.markBg = rec.getValue(GUILDMARK.MARKBG);
        this.mark = rec.getValue(GUILDMARK.MARK);
        this.markBgColor = rec.getValue(GUILDMARK.MARKBGCOLOR);
        this.markColor = rec.getValue(GUILDMARK.MARKCOLOR);
    }

    public GuildMark(short markBg, short mark, byte markBgColor, byte markColor) {
        this.markBg = markBg;
        this.mark = mark;
        this.markBgColor = markBgColor;
        this.markColor = markColor;
    }

    public void encode(PacketWriter pw) {
        pw.writeShort(markBg);
        pw.write(markBgColor);
        pw.writeShort(mark);
        pw.write(markColor);
    }
}

File: src/main/java/world/guild/GuildMark.kt
package world.guild

import database.jooq.Tables
import org.jooq.Record
import util.packet.PacketReader
import util.packet.PacketWriter

class GuildMark {

    val markBg: Short
    val mark: Short
    val markBgColor: Byte
    val markColor: Byte

    constructor(reader: PacketReader) {
        markBg = reader.readShort()
        markBgColor = reader.readByte()
        mark = reader.readShort()
        markColor = reader.readByte()
    }

    constructor(rec: Record) {
        markBg = rec.getValue(Tables.GUILDMARK.MARKBG)
        mark = rec.getValue(Tables.GUILDMARK.MARK)
        markBgColor = rec.getValue(Tables.GUILDMARK.MARKBGCOLOR)
        markColor = rec.getValue(Tables.GUILDMARK.MARKCOLOR)
    }

    constructor(markBg: Short, mark: Short, markBgColor: Byte, markColor: Byte) {
        this.markBg = markBg
        this.mark = mark
        this.markBgColor = markBgColor
        this.markColor = markColor
    }

    fun encode(pw: PacketWriter) {
        pw.writeShort(markBg)
        pw.write(markBgColor.toInt())
        pw.writeShort(mark)
        pw.write(markColor.toInt())
    }
}